src/tic_tac_toe/GameBoard.java

```diff
@@ -32,8 +32,11 @@ public void playingGame() {
             printBoard();
             System.out.println("Hey user, pick a number between 1-9");
             String usersMove = input.nextLine();
-            while (!checkIfValidMove(usersMove, board)) {
-                System.out.println("Please enter a valid number");
+            while (!checkIfValidMove(usersMove, board) | !checkIfMoveTaken(usersMove, board)) {
+                if (usersMove.isEmpty()) System.out.println("You did not select a move, please enter a number");
+                else if(!checkIfNumber(usersMove)) System.out.println("Only numbers can be accepted as valid moves");
```

Comment:
Niceeeee! I see you're doing guard clauses. Did they teach you about guard clauses or is this just you being smart af

Reply:
I don't even know what a guard clause is lol

Reply:
It's a clean code practice. Instead of having a bunch of nested if statements that are three four deep, you do the negate condition. 

https://medium.com/@scadge/if-statements-design-guard-clauses-might-be-all-you-need-67219a1a981a